File: demo_communication_center/app/src/main/java/com/bandyer/demo_communication_center/adapter_items/UserItem.java
/*
 * Copyright (C) 2018 Bandyer S.r.l. All Rights Reserved.
 * See LICENSE.txt for licensing information
 */

package com.bandyer.demo_communication_center.adapter_items;

import android.support.annotation.NonNull;
import android.view.View;
import android.widget.TextView;

import com.bandyer.demo_communication_center.R;
import com.mikepenz.fastadapter.items.AbstractItem;

import java.util.List;

import butterknife.BindView;

/**
 * A simple RecyclerView item used to display the user name as a cell in the list.
 */
public class UserItem extends AbstractItem<UserItem, UserItem.ViewHolder> {

    public final String name;

    public UserItem(String name) {
        this.name = name;
    }

    @Override
    public int getType() {
        return R.id.user_item_id;
    }

    @Override
    public int getLayoutRes() {
        return R.layout.item_user;
    }

    @NonNull
    @Override
    public ViewHolder getViewHolder(@NonNull View v) {
        return new ViewHolder(v);
    }

    protected static class ViewHolder extends ButterKnifeViewHolder<UserItem> {

        @BindView(R.id.user)
        TextView name;

        ViewHolder(View view) {
            super(view);
        }

        @Override
        public void bindView(@NonNull UserItem item, @NonNull List<Object> payloads) {
            name.setText(item.name);
        }

        @Override
        public void unbindView(@NonNull UserItem item) {
            name.setText(null);
        }
    }
}

File: demo_communication_center/app/src/main/java/com/bandyer/demo_communication_center/adapter_items/UserItem.kt
/*
 * Copyright (C) 2018 Bandyer S.r.l. All Rights Reserved.
 * See LICENSE.txt for licensing information
 */

package com.bandyer.demo_communication_center.adapter_items

import android.view.View
import com.bandyer.demo_communication_center.R
import com.bandyer.demo_communication_center.databinding.ItemUserBinding
import com.mikepenz.fastadapter.FastAdapter
import com.mikepenz.fastadapter.items.AbstractItem

/**
 * A simple RecyclerView item used to display the user alias as a cell in the list.
 */
class UserItem(var userAlias: String) : AbstractItem<UserItem.ViewHolder>() {

    override val layoutRes: Int
        get() = R.layout.item_user

    override val type: Int
        get() = R.id.user_item_id

    override fun getViewHolder(v: View) = ViewHolder(v)

    class ViewHolder(view: View) : FastAdapter.ViewHolder<UserItem>(view) {

        private val binding: ItemUserBinding = ItemUserBinding.bind(view)

        override fun bindView(item: UserItem, payloads: List<Any>) {
            binding.userAlias.text = item.userAlias
        }

        override fun unbindView(item: UserItem) {
            binding.userAlias.text = null
        }
    }
}